payment/datastore.go

```diff
@@ -350,9 +353,73 @@ func (pg *Postgres) GetTransaction(externalTransactionID string) (*Transaction,
 	return &transaction, nil
 }
 
+// CheckExpiredCheckoutSession - check order metadata for an expired checkout session id
+func (pg *Postgres) CheckExpiredCheckoutSession(orderID uuid.UUID) (bool, string, error) {
+	var (
+		expired         bool
+		checkoutSession string
+		err             error
+	)
+
+	err = pg.RawDB().Get(&checkoutSession, `
+		SELECT metadata->>'stripeCheckoutSessionId'
+		FROM orders
+		WHERE id = $1 
+			AND metadata is not null
+			AND status='pending'
+			AND updated_at<now() - interval '1 hour'
+	`, orderID)
+
+	if err == nil && checkoutSession != "" {
+		expired = true
+	}
+	if errors.Is(err, sql.ErrNoRows) {
+		// if there are no rows, then we are not expired
+		// drop this error
+		return expired, checkoutSession, nil
+	}
+	return expired, checkoutSession, err
+}
+
+func (pg *Postgres) isStripeSub(orderID uuid.UUID) (bool, string, error) {
+	var (
+		ok  bool
+		md  datastore.Metadata
+		err error
+	)
+
+	err = pg.RawDB().Get(&md, `
+		SELECT metadata
+		FROM orders
+		WHERE id = $1 AND metadata is not null
+	`, orderID)
+
+	if err == nil {
+		if v, ok := md["stripeSubscriptionId"]; ok {
+			return ok, v, err
+		}
+	}
+	return ok, "", err
+}
+
 // UpdateOrder updates the orders status.
 // 	Status should either be one of pending, paid, fulfilled, or canceled.
 func (pg *Postgres) UpdateOrder(orderID uuid.UUID, status string) error {
+	// if the order is cancelled we might need to inform other systems
+	if status == "canceled" {
```

Comment:
it seems strange to have this logic live in the db, can we instead have a non-handler cancel method on the service which both updates the db and performs this?

Reply:
Can we make `canceled` an enum, as `canceled` and `cancelled` are both valid English words?

Reply:
what's an enum, this is golang

Reply:
resolved both comments: https://github.com/brave-intl/bat-go/pull/889/commits/df62ffdda5dfffd1eea14232563c78228af9d6c9